plugins/browser-plugin-button-click-tracking/src/api.ts

```diff
@@ -0,0 +1,109 @@
+import { BrowserPlugin, BrowserTracker, dispatchToTrackersInCollection } from '@snowplow/browser-tracker-core';
+import { DynamicContext, CommonEventProperties, resolveDynamicContext } from '@snowplow/tracker-core';
+
+import { Filter, FilterFunction, createEventFromButton, filterFunctionFromFilter } from './util';
+import { ButtonClickEvent, buildButtonClick } from './core';
+
+/** The configuration for automatic button click tracking */
+export interface ButtonClickTrackingConfiguration {
+  /** The filter options for the button click tracking */
+  filter?: Filter;
+  /** The dynamic context which will be evaluated for each button click event */
+  context?: DynamicContext;
+}
+
+const _trackers: Record<string, BrowserTracker> = {};
+
+/**
+ * Button click tracking
+ *
+ * Will automatically tracking button clicks once enabled with 'enableButtonClickTracking'
+ * or you can manually track button clicks with 'trackButtonClick'
+ */
+export function ButtonClickTrackingPlugin(): BrowserPlugin {
+  return {
+    activateBrowserPlugin: (tracker: BrowserTracker) => {
+      _trackers[tracker.id] = tracker;
+    },
+  };
+}
+
+/**
+ * Manually log a click
+ *
+ * @param event - The event information
+ * @param trackers - The tracker identifiers which the event will be sent to
+ */
+export function trackButtonClick(
+  event: ButtonClickEvent & CommonEventProperties,
+  trackers: Array<string> = Object.keys(_trackers)
+) {
+  dispatchToTrackersInCollection(trackers, _trackers, (t) => {
+    t.core.track(buildButtonClick(event), event.context, event.timestamp);
+  });
+}
+
+/**
+ *
+ * Enable automatic button click tracking
+ *
+ * @param configuration  - The configuration for automatic button click tracking
+ * @param trackers - The tracker identifiers which the event will be sent to
+ */
+export function enableButtonClickTracking(
```

Comment:
What it be too much of a hassle to add `disableButtonClickTracking`?
With SPAs people are now more keen to enable and disable tracker features only for a certain period of the page lifecycle. We have had to add a disable activity tracking feature recently for instance.

Reply:
Added!